packages/pyright-internal/src/analyzer/checker.ts

```diff
@@ -1534,6 +1534,18 @@ export class Checker extends ParseTreeWalker {
             }
         }
 
+        if (
+            reportAsUnused &&
+            this._fileInfo.ipythonMode === IPythonMode.CellDocs &&
+            node.parent?.nodeType === ParseNodeType.StatementList &&
+            node.parent.parent?.nodeType === ParseNodeType.Module &&
+            node.parent.statements[node.parent.statements.length - 1] === node
```

Comment:
Looks like there's something wrong with this check, since the following generates zero errors. I would expect the first line to still get an error.

```python
4
4
```

Reply:
Needed to check that the expression's parent `StatementList` is the last statement in the module.